File: src/node/magma/JavaFiles.ts





class JavaFiles {
	constructor () {
	}
	walk(root : Path) : Result<List<Path>, IOException>;
	writeString(path : Path, output : CharSequence) : Optional<IOException>;
	createDirectories(path : Path) : Optional<IOException>;
	readString(source : Path) : Result<string, IOException>;
}

File: src/java/magma/JavaFiles.java
package magma;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

public class JavaFiles {
    private JavaFiles() {
    }

    @Actual
    static Result<List<Path>, IOException> walk(final Path root) {
        try (final var stream = Files.walk(root)) {
            return new Ok<>(stream.toList());
        } catch (final IOException e) {
            return new Err<>(e);
        }
    }
}
